draft-fossati-tls-exported-attestation.md

```diff
@@ -72,38 +86,119 @@ There is a growing need to demonstrate to a remote party that cryptographic keys
 
 The reader is assumed to be familiar with the vocabulary and concepts defined in RFC 9334 and RFC 9261.
 
-We use the term REMOTE_ATTESTATION payload to refer to the opaque token generated by the TLS Exported Authenticator implementation. The content is opaque to the application layer protocol but, of course, not to the TLS Exported Authenticator implementation.
+# cmw_attestation Extension
+
+It introduces a new TLS extension, cmw_attestation, which enables the inclusion of attestation information directly in the Certificate message. This approach eliminates the need for real-time certificate issuance from a Certificate Authority (CA) and minimizes handshake delays. Typically, CAs require several seconds to minutes to issue a certificate due to verification steps such as validating subject identity, signing the certificate, and distributing it. These delays introduce latency into the TLS handshake, making real-time certificate generation impractical. The cmw_attestation extension circumvents this issue by embedding attestation data within the Certificate message itself, removing reliance on external certificate issuance processes.
+
+The `cmw_attestation` extension is defined to be included only in the Certificate message during the Exported Authenticator-based post-handshake authentication. This ensures that attestation evidence is conveyed within the Certificate message without requiring modifications to the X.509 certificate structure.
+
+~~~ aasvg
+struct {
+    opaque cmw_data<1..2^16-1>;
+} CMWAttestationExtension;
+~~~
+
+cmw_data: Encapsulates the attestation evidence in a format compatible with CMW. The cmw_data field MUST be encoded using CBOR or JSON (as per {{I-D.ietf-rats-msg-wrap}}).
+
+
+## Negotiation of CMWAttestationExtension
+
+Clients and servers use the TLS flags extension defined in {{I-D.ietf-tls-tlsflags}} to indicate support for the functionality defined in this document. We refer to the previously defined "cmw_attestation" extension,
+and the corresponding flag is called the "CMW_Attestation" flag.
+
+The "CMW_Attestation" flag proposed by the client in the ClientHello MUST be acknowledged in the EncryptedExtensions if the server also  supports the functionality defined in this document and is configured to use it.
+
+If the "CMW_Attestation" flag is not set, servers ignore any of the functionality specified in this document, and attestation evidence cannot be conveyed using "Exported TLS Authenticators".
+
+## Usage in Post-Handshake Authentication
+
+The `cmw_attestation` extension is designed to be used exclusively in post-handshake authentication as defined in {{RFC9261}}. It allows attestation evidence to be transmitted in the authenticator (Certificate) message only in response to an authenticator request. This ensures that attestation data is provided on demand rather than being included in the initial TLS handshake.
```

Comment:
This line specifically calls out attestation evidence instead of attestation information (like in line 110). Is this on purpose? I can see that line 136 also carries on with background check only.

Note: maybe we can keep a consistent terminology and use `remote attestation credentials` or `attestation credentials`, like the other draft. "Attestation information" seems a bit more vague.

Reply:
Thanks, updated to use "attestation credentials".